CMakeLists.txt

```diff
@@ -969,6 +969,7 @@ endif()
 # CUDA features
 ################################################################################
 if(HPX_WITH_CUDA AND NOT HPX_WITH_CUDA_CLANG)
+  hpx_add_target_compile_option(-DHPX_WITH_CUDA)
```

Comment:
This is wrong. The actual problem lies here: https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/config/compiler_specific.hpp#L75 which should test for `HPX_HAVE_CUDA` instead.